### PR TITLE
docs: add `why-we-need-apache-apisix` blog

### PR DESCRIPTION
**Pull request by guitu168**
because `blog/en/blog/2021/06/28/why-we-need-Apache-APISIX.md` has a lot of traffic, but the content is too verbose, I re-written a document and deleted the original document. Update the Chinese version, and the English version will be updated later.

Changes:

Screenshots of the change:

### CODE REVIEW

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+下图为 APISIX 的生态图，从该图可以准确看到 APISIX 已经支持了 7 层协议有 HTTP(S)、HTTP2、Dubbo、QUIC 和物联网协议 MQTT 等等，4 层协议有 TCP/UDP。
+右侧部分则是一些开源或者 SaaS 服务，比如 SkyWalking、Prometheus 、Vault 等等。而最下面则是比较常见的操作系统环境、云厂商和硬件环境。而作为一个开源软件，APISIX 也支持在 ARM64 的服务器上运行。
+
+![APISIX's Ecosystem](https://static.apiseven.com/2022/blog/0729/2.PNG)
+
+APISIX 不仅支持众多协议与操作系统，而且也支持多语言编程插件。APISIX 诞生之初仅支持使用 Lua 语言编写插件，这种情况下就需要开发者掌握 Lua 和 NGINX 相关的技术栈。然而 Lua 和 NGINX 属于相对小众的技术，开发者很少。因此我们在 APISIX 上支持了多语言开发插件，目前已经正式支持了 Java、Golang、Node.js、Python 等语言。
+
+![programming language](https://static.apiseven.com/2022/blog/0729/3.png)
+
+活跃的社区
```
Comment (yzeng25) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
### 活跃的社区
```

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+下图是贡献者增长曲线，其中横坐标代表时间线，纵坐标代表贡献者总数。我们可以看到 APISIX 和 Kong 这两个项目相对更活跃，APISIX 的增长速度从开源第一天就保持着非常不错的增长率，在以接近 Kong 两倍的速度快速成长，并且贡献者数量已经超过了 Kong，由此可见 APISIX 受欢迎程度。当然评价一个项目活跃度还有很多其他方法，比如查看每月活跃 Issue、PR 总数等方式，值得高兴的是 APISIX 在这些方面也是一骑绝尘。
+
+![Contributor graph](https://static.apiseven.com/2022/blog/0729/4.png)
+
+## APISIX 应用场景
```
Comment (yzeng25) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
## APISIX 的应用场景
```

**Review by yzeng25 — COMMENTED** [review]
Nice work!

**Comment by yzeng25** [comment]
I suggest keeping the original file path `./2021/06/28/why-we-need-apache-apisix`, otherwise people get a default 404 page when they visit.

**Comment by guitu168** [comment]
> I suggest keeping the original file path `./2021/06/28/why-we-need-apache-apisix`, otherwise people get a default 404 page when they visit.

I'm not sure, maybe I can add a redirect to fix the 404. cc @yzeng25

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+而 [APISIX Ingress Controller](https://apisix.apache.org/zh/docs/ingress-controller/getting-started/) 则完美解决了上面提到的所有问题：支持全动态，无需重启加载。同时继承了 APISIX 的所有优势，还支持原生 Kubernetes CRD，方便用户迁移。
+
+![Apisix Kubernetes Ingress](https://static.apiseven.com/2022/blog/0729/6.png)
+
+### 服务网格
+
+未来五到十年，基于云原生模式架构下的服务网格架构开始崭露头角。APISIX 也提前开始锁定赛道，通过调研和技术分析后，APISIX 已经支持了 xDS 协议，APISIX Mesh 就此诞生，在服务网格领域  APISIX 也拥有了一席之地。
+
+![apisix mesh](https://static.apiseven.com/2022/blog/0729/7.png)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
![APISIX Mesh](https://static.apiseven.com/2022/blog/0729/7.png)
```

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+APISIX 目前支持多种语言编写扩展插件，可以解决东西向微服务 API 网关面临的主要问题——异构多语言和通用问题。内置支持的服务注册中心有 Nacos、etcd、Eureka 等，还有标准的 DNS 方式，可以平滑替代 Zuul、Spring Cloud Gateway、Dubbo 等微服务 API 网关。
+
+### Kubernetes Ingress
+
+目前 K8s 官方 Kubernetes Ingress Controller 项目主要基于 NGINX 配置文件的方式，所以在路由能力和加载模式上稍显不足，并且存在一些明显劣势。比如添加、修改任何 API 时，需要重启服务才能完成新 NGINX 配置的更新，但重启服务，对线上流量的影响是非常大的。
+
+而 [APISIX Ingress Controller](https://apisix.apache.org/zh/docs/ingress-controller/getting-started/) 则完美解决了上面提到的所有问题：支持全动态，无需重启加载。同时继承了 APISIX 的所有优势，还支持原生 Kubernetes CRD，方便用户迁移。
+
+![Apisix Kubernetes Ingress](https://static.apiseven.com/2022/blog/0729/6.png)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
![APISIX Kubernetes Ingress](https://static.apiseven.com/2022/blog/0729/6.png)
```

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+下图是贡献者增长曲线，其中横坐标代表时间线，纵坐标代表贡献者总数。我们可以看到 APISIX 和 Kong 这两个项目相对更活跃，APISIX 的增长速度从开源第一天就保持着非常不错的增长率，在以接近 Kong 两倍的速度快速成长，并且贡献者数量已经远远超过了 Kong，由此可见 APISIX 受欢迎程度。当然评价一个项目活跃度还有很多其他方法，比如查看每月活跃 Issue、PR 总数等方式，值得高兴的是 APISIX 在这些方面也是一骑绝尘。
+
+![Contributor graph](https://static.apiseven.com/2022/blog/0729/4.png)
+
+## APISIX 的应用场景
+
+从下图中，相信你已经看出 APISIX 的目标：**统一代理基础设施**。
+
+![apisix Application scenarios](https://static.apiseven.com/2022/blog/0729/5.png)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
![APISIX application scenarios](https://static.apiseven.com/2022/blog/0729/5.png)
```

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+而 APISIX 的出现则解决了上述所有问题，成为了云原生时代最完美的 API 网关。那么 Apache APISIX 的优势到底是什么？为什么可以在短短三年的时间里成为全世界最活跃的网关？
+
+## APISIX 的优势
+
+### 优异的架构
+
+首先 Apache APISIX 拥有优异的架构，现在很多应用都在向微服务、容器化迁移，形成新的云原生时代。云原生作为当下的技术潮流，将重写传统企业的技术架构。而 APISIX 自诞生之初就跟随技术潮流，并将其设计为云原生架构：
+
+![apisix](https://static.apiseven.com/2022/blog/0729/1.png)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md]:
```suggestion
![APISIX](https://static.apiseven.com/2022/blog/0729/1.png)
```

File: blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
+---
+title: "有了 NGINX 和 Kong，为什么还需要 Apache APISIX？"
+author: "韩飞"
+authorURL: "https://github.com/hf400159"
+authorImageURL: "https://github.com/hf400159.png"
```
Comment (yzeng25) [inline blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md:5]: need to add credit for yuansheng as well

Comment (guitu168) [reply blog/zh/blog/2022/07/29/why-we-need-apache-apisix.md:5]: This is a new article.

**Review by SylviaBABY — APPROVED** [review]

**Review by yzeng25 — APPROVED** [review]

**Review by Baoyuantop — APPROVED** [review]

**Pull request merged**